### DISCUSSION TITLE
neo-tree keymap

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#1923] [Q&A] [answered]
I want to change some keymaps for `neo-tree`.
I do some google work and read the source code of `neo-tree`. And got the script like this
```lua
return {
  "nvim-neo-tree/neo-tree.nvim",
  config = function(_, config)
    config.window.mappings = vim.tbl_extend("force", config.window.mappings, {
      ["l"] = "open",
      ["h"] = "close_node",
    })
  end,
}
```

But it is not work why. How to fix it?

**Comment by dpetka2001** [comment, marked as answer]
Do it in the `opts`, not the `config`. Use the `config` if you want to change the behavior of the default `setup` function. Take a look [here](https://github.com/dpetka2001/dotfiles/blob/9a9fe1ae78e2f1cb61a08051d54bb5433d6346d1/dot_config/nvim/lua/plugins/editor.lua#L4-L30).

**Reply by OriLiMu** [reply]
> Do it in the `opts`, not the `config`. Use the `config` if you want to change the behavior of the default `setup` function. Take a look [here](https://github.com/dpetka2001/dotfiles/blob/9a9fe1ae78e2f1cb61a08051d54bb5433d6346d1/dot_config/nvim/lua/plugins/editor.lua#L4-L30).
>

Another question please I see your code is 
```lua
    window = {
      mappings = {
        ["l"] = "open",
        ["h"] = "close_node",
      },
    },
```
You don't use the `tbl_extend` which will merge the config option manually. 
So the lazy nvim will merge the `opts` by default right?

**Reply by dpetka2001** [reply]
Yes, LazyVim merges the values in `opts` by default. See `lazy.nvim` [docs](https://github.com/folke/lazy.nvim#%EF%B8%8F-importing-specs-config--opts) and `LazyVim` [docs](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-customizing-plugin-specs) to understand which values get merged.
